### PR TITLE
test: client and logout ap test

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

#98  

### Solutions

test the request and the fail 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

test api request configuration
 
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/user/client/RegisterClientRequest.kt [hunk 1]
```diff
@@ -7,53 +7,95 @@ import kotlinx.serialization.Serializable
...
+        }
+    },
 
     @SerialName("legalhold")
-    LegalHold
+    LegalHold {
+        override fun toString(): String {
+            return "legalhold"
+        }
+    }
```
Comment (vitorhugods) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/user/client/RegisterClientRequest.kt:90]: What is the `toString` necessary for?

Maybe can we just override the enum class `toString`?

```kotlin
enum class Foo {
BAR, FOOBAR;

    override fun toString(): String {
        return this.name.lowercase()
    }
}
```

File: network/src/commonTest/kotlin/com/wire/kalium/api/tools/json/api/user/client/RegisterClientResponseJson.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        """.trimMargin()
+    }
+
+    val valid = ValidJsonProvider(
+        RegisterClientResponse(
+            clientId = "defkrr8e7grgsoufhg8",
+            type = ClientType.Permanent,
+            deviceType = DeviceType.Phone,
+            registrationTime = "2021-05-12T10:52:02.671Z",
+            location = LocationResponse(latitude = "1.2345", longitude = "6.7890"),
```
Comment (vitorhugods) [inline network/src/commonTest/kotlin/com/wire/kalium/api/tools/json/api/user/client/RegisterClientResponseJson.kt:41]: Really close to null island. Other than that, no easter eggs in [this location](https://www.google.com/maps/place/1°14'02.4"N+5°40'40.8"E/@0.8699224,5.179952,6.88z).

**Review by vitorhugods — APPROVED** [review]
Looks good

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/user/client/RegisterClientRequest.kt (hunk 1 above)
Comment (MohamadJaara) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/user/client/RegisterClientRequest.kt:90]: it's for testing purposes , i guess override the enum class toString will work in most cases
i will change it

**Pull request merged**